go/Makefile.common

```diff
@@ -6,6 +6,12 @@ VERSION        := 1.23.5
 DEBIAN_VERSION ?= 9
 SUFFIX         := -$(shell basename $(CURDIR))
 TAG_EXTENSION  ?=
+FIPS           ?=
+ifeq "${FIPS}" "true"
+ifeq ($(findstring -fips,$(TAG_EXTENSION)),)
+TAG_EXTENSION:=$(TAG_EXTENSION)-fips
```

Comment:
Can you try disabling the cross compilation for FIPS via also setting `BUILDX:=0`  here? 
This fixed the issue for me when running `.buildkite/scripts/build.sh Makefile` in a small VM.